File: src/code-samples/deepagents/async-subagents-hybrid.ts
// :snippet-start: async-subagents-hybrid-js
import type { AsyncSubAgent } from "deepagents";

const asyncSubagents: AsyncSubAgent[] = [
  {
    name: "researcher",
    description: "Research agent",
    graphId: "researcher",
    // No url → ASGI (co-deployed)
  },
  {
    name: "coder",
    description: "Coding agent",
    graphId: "coder",
    url: "https://coder-deployment.langsmith.dev",
    // url present → HTTP (remote)
  },
];
// :snippet-end:

// :remove-start:
if (asyncSubagents.length !== 2) {
  throw new Error("expected two async subagents");
}
console.log("✓ async-subagents-hybrid sample validated");
// :remove-end:

File: src/code-samples/deepagents/async-subagents-hybrid.py
"""Async subagents: hybrid ASGI and HTTP deployment."""

# :snippet-start: async-subagents-hybrid-py
from deepagents import AsyncSubAgent

async_subagents = [
    AsyncSubAgent(
        name="researcher",
        description="Research agent",
        graph_id="researcher",
        # No url → ASGI (co-deployed)
    ),
    AsyncSubAgent(
        name="coder",
        description="Coding agent",
        graph_id="coder",
        url="https://coder-deployment.langsmith.dev",
        # url present → HTTP (remote)
    ),
]
# :snippet-end:

# :remove-start:
assert len(async_subagents) == 2
print("✓ async-subagents-hybrid sample validated")
# :remove-end:
